openweb/thematic/clusters.py

```diff
@@ -0,0 +1,78 @@
+"""
+Module to do multi-resolution clustering.
+"""
+from typing import Iterable, List, cast
+
+import numpy as np
+
+from openweb.thematic.third_party import community_detection
+
+
+class FastCommunityDetection:
+    """sentence_transformers fast community detection wrapped with sklearn interace."""
+
+    def __init__(self, threshold=0.75, min_community_size=10, init_max_size=1000):
+        self._kwargs = {
+            "threshold": threshold,
+            "min_community_size": min_community_size,
+            "init_max_size": init_max_size,
+        }
+        self.labels_: np.ndarray = np.array([])
+        self.cluster_centers_: np.ndarray = np.ndarray([])
+
+    def fit(
+        self,
+        X,
+        y=None,  # pylint: disable=unused-argument
+    ) -> "FastCommunityDetection":
+        self.labels_ = np.repeat(-1, len(X))
+        cluster_centers = cast(List[int], [])
+        clusters = community_detection(X, **self._kwargs)
+        for cluster_id, cluster in enumerate(clusters):
+            cluster_centers.append(X[cluster[0]])
```

Comment:
I'm assuming `cluster_centers` is the cluster embedding centroid of the detected community? If that's the case, why assign the first member embedding of the community as the centroid, versus say taking the mean of all the members?

Reply:
wrt to above issue, reason is because in the original implementation of `community_detection` algorithm in sentence-transformers library, the first member is the anchor node against which the other datapoints are grouped around. can consider giving option to override this behavior in future implementation.